### PR TITLE
docs(holochain-types): fix warnings

### LINKED ISSUE #4540: holochain_types 9 warnings

### PR DESCRIPTION
**Pull request by jost-s**
### Summary
resolves #4540 

### TODO:
- [ ] CHANGELOGs updated with appropriate info
- [ ] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/holochain_types/src/app.rs [hunk 1]
```diff
@@ -144,7 +144,7 @@ pub struct InstallAppPayload {
     /// agent keys if you lose access to the device with your conductor data
     /// (as long as you retain exclusive access to the device seed of course).
     ///
-    /// Holochain will only generate an agent key for you if [`ConductorConfig::device_seed_lair_tag`]
+    /// Holochain will only generate an agent key for you if a device seed tag
```
Comment (jost-s) [inline crates/holochain_types/src/app.rs:147]: Not available in this crate

File: crates/holochain_types/src/app.rs [hunk 2]
```diff
@@ -212,8 +212,8 @@ pub enum RoleSettings {
     Provisioned {
         /// When the app being installed has the `allow_deferred_memproofs` manifest flag set,
         /// passing `None` for this field for all roles in the app will allow the app to enter
-        /// the "deferred membrane proofs" state, so that memproofs can be provided later via
-        /// [`AppRequest::ProvideMemproofs`]. If `Some` is used here, whatever memproofs are
```
Comment (jost-s) [inline crates/holochain_types/src/app.rs:216]: conductor api types not available in types

File: crates/holochain_types/src/app.rs [hunk 3]
```diff
@@ -963,8 +963,8 @@ pub enum AppStatus {
     /// and will not restart automaticaly on conductor reboot.
     Disabled(DisabledAppReason),
 
-    /// The app is installed, but genesis has not completed due to use of
-    /// [`MembraneProofProvisioning::Deferred`]
+    /// The app is installed, but genesis has not completed because Membrane Proofs
+    /// have not been provided.
```
Comment (jost-s) [inline crates/holochain_types/src/app.rs:967]: not available

File: crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs [hunk 4]
```diff
@@ -40,8 +40,8 @@ pub struct AppManifestV1 {
 
     /// Declares that the app may be installed without the need to
     /// specify membrane proofs at installation time. If memproofs are not
-    /// provided at install time, they must be provided later by the
-    /// [`AppRequest::ProvideMemproofs`] call. If memproofs are provided
+    /// provided at install time, they must be provided later before the
+    /// app can be enabled. If memproofs are provided
```
Comment (jost-s) [inline crates/holochain_types/src/app/app_manifest/app_manifest_v1.rs:44]: n/a

File: crates/holochain_types/src/countersigning.rs [hunk 5]
```diff
@@ -68,7 +68,7 @@ pub enum CountersigningSessionState {
     ///
     /// Note that because the [PreflightRequest] is stored here, we only ever enter the unknown state
     /// if we managed to keep the preflight request in memory, or if we have been able to recover it
-    /// from the source chain as part of the committed [CounterSigningSessionData]. Otherwise, we
+    /// from the source chain as part of the committed countersigning session data. Otherwise, we
```
Comment (jost-s) [inline crates/holochain_types/src/countersigning.rs:71]: not available here

File: crates/holochain_types/src/countersigning.rs [hunk 6]
```diff
@@ -122,8 +122,6 @@ pub struct SessionResolutionSummary {
     pub required_reason: ResolutionRequiredReason,
     /// How many attempts have been made to resolve the session.
     ///
-    /// Attempts are made according to the frequency specified by [RETRY_UNKNOWN_SESSION_STATE_DELAY].
```
Comment (jost-s) [inline crates/holochain_types/src/countersigning.rs:125]: That's no longer a const and not available here.

**Review by neonphog — APPROVED** [review]

**Pull request merged**